sycl/include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp

```diff
@@ -339,6 +378,9 @@ struct joint_matrix_store_impl<
     } else if constexpr (std::is_same<T, double>::value) {
       __dmma_m8n8k4_st_c_f64(dst.get(), src.data, stride,
                              get_layout_id<Layout>());
+    } else if constexpr (std::is_same<T, int32_t>::value) {
+      __bmma_m8n8k128_st_c_i32(dst.get(), src.data, stride,
```

Comment:
what is the difference between  __imma_m32n8k16_st_c_i32 and __bmma_m8n8k128_st_c_i32 other than the shape difference? 
Does "b" stand for binary here and it stores only results of bmad?
For my education, do you know how  K is so big in the bmad case (128)?

Reply:
Yes "b" is for binary and it can only be used with the bmad case.  Since the shapes are different the two instructions load different number of registers/fragments for storing the C/Accumulator matrix type.  So it is important that there is the two distinct instructions in this case.

So k=128 because it refers to the number of bits in the K dimension which is equal to the number of elements in the K dimension for the bitwise case.  The number following k in these instructions always refers to a number of matrix elements, which may or may not equal the number of registers used to store these elements, depending on how the matrix elements are packed.  For the bmad cases we store 32 bit elements into a single int32_t register. For bf16 cases we store 2 elements into a single int32_t etc.

Reply:
In a related note it is worth noticing that there are two of these C load/store instructions e.g. `__mma_tf32_m16n16k8_ld_c` and `__hmma_m16n16k16_ld_c_f32` which when compiled down to the lower level assembly (SASS) for all compatible sm versions result in the exact same instructions!  We plan to take advantage of this to use the standard matrix extension interface for the tf32 type, which otherwise would require a new interface specialization for Nvidia backend.

Reply:
>>> So k=128 because it refers to the number of bits in the K dimension which is equal to the number of elements in the K dimension for the bitwise case. 

In the test case, this writing: joint_matrix<uint32_t, matrix_use::a, 8, 128,  matrix_layout::row_major> means
that 128 refers to number of uint32_t elements in a row of matrix A. How this is the same as number of bits as number of bits would be 128*32?

Reply:
I've now updated https://github.com/intel/llvm-test-suite/pull/760; I've renamed A/B as A_Packed/B_Packed, added comments making it clear that each array element of A_Packed contains 32 single-bit matrix elements.

Reply:
I think the implementation should change to have this in user code:
 joint_matrix<uint32_t, matrix_use::a, 8, 128/32, matrix_layout::row_major>;
bmad intrinsics (or the implementation) work on number of bits but users should not.
In the header file here, you should multiply by 32 to retrieve the number of bits to pass to the intrinsics rather than having divide by 32.
From the user perspective, they should operate on number of elements (not bits).

Reply:
You're probably right.  I've made the changes you describe. Thanks